### ISSUE TITLE
needs.json schema? How are links stored?

### ISSUE THREAD

**Issue by arwedus** [useblocks/sphinx-needs#1145]
Is it currently correct to store custom link options of a need flat, e.g. 

```json
                "ACTIVITY_ThisAndThat": {
                    "description": "",
                    "id": "ACTIVITY_ThisAndThat",
                    "implements": [],
                    "has_in_port": ["...", ...],
                    "has_out_port": ["...", ...],
                    "instantiates": ["...", ...]
                 },
```

or nested below "links", like this:

```json
                "ACTIVITY_ThisAndThat": {
                    "description": "",
                    "id": "ACTIVITY_ThisAndThat",
                    "implements": [],
                    "links": {
                        "has_in_port": "...,...",
                        "has_out_port": "...",
                        "instantiates": "...",
                    },
                }
```

I could not find anything in the docs. I'm writing a needs.json generator right now, so I needs to know :laughing:

**Comment by danwos** [comment]
The first one is correct, so it's just a "flat" dictionary.

**Comment by chrisjsewell** [comment]
I think this can be closed with #1230
